src/open_inwoner/accounts/views/contacts.py

```diff
@@ -125,3 +130,34 @@ def post(self, request, *args, **kwargs):
 
         self.log_change(object, _("contact relationship was removed"))
         return HttpResponseRedirect(self.success_url)
+
+
+class ContactApprovalView(LogMixin, LoginRequiredMixin, SingleObjectMixin, View):
+    model = User
+    slug_field = "uuid"
+    slug_url_kwarg = "uuid"
+    success_url = reverse_lazy("accounts:contact_list")
+
+    def get_queryset(self):
+        current_user = self.request.user
+        base_qs = User.objects.get_pending_approvals(current_user)
+        return base_qs
+
+    def post(self, request, *args, **kwargs):
+        sender = self.get_object()
+        receiver = self.request.user
+        approved = request.POST.get("contact_approve")
+        rejected = request.POST.get("contact_reject")
+        if approved or rejected:
+            self.update_contact(sender, receiver, (approved or rejected))
+            return HttpResponseRedirect(self.success_url)
+
+    def update_contact(self, sender, receiver, type_of_approval):
+        if type_of_approval == "approve":
+            sender.contacts_for_approval.remove(receiver)
+            sender.user_contacts.add(receiver)
+            self.log_change(sender, _("contact was approved"))
+
+        elif type_of_approval == "reject":
```

Comment:
I think it is convention to not use raw POST but always try to go for a Form, and use Choices instead of magic strings so validation is a bit harder and more regular. 

The two buttons can have the same name in the HTML but different values and then the Form is just a CharField with the choices for the incoming data.

Also, looking at the post method: what happens if there is no approved or rejected? It could `raise BadRequest` since there is no HTML for this view.

Reply:
Discussed this with Anna as well and I decided to go for the same approach as the one in deleting a contact. That's why you saw the ids in the for loop.

Reply:
Ok, if it's same pattern as the other and Anna is ok with it then it's fine.

Reply:
Let's wait for @annashamray 's approval as well because she hasn't seen this implemented.

Reply:
I think there is a little misunderstanding. 
I said that there is no need in adding additional forms into `ContactListView` context - we can just generate forms in the template since it's only csrf token, two buttons and 1 action. It's how the delete buttons for contacts are done. 
Security-wise it should be ok, since we send csrf token. 

But it's a different story with the view which process POST request (`ContactApprovalView`). 
Here we can use forms to escape and validate user input, especially if we use values in buttons. We can use only button names like django admin does, and they do it without extra validation. I don't know is it a best practice or not.